web-common/src/lib/event-bus/events.ts

```diff
@@ -16,6 +16,17 @@ interface NotificationOptions {
 }
 
 export interface BannerMessage {
+  type: "default" | "success" | "info" | "warning" | "error";
   message: string;
   includesHtml?: boolean;
+
+  iconType: "none" | "alert" | "check" | "sleep" | "loading";
+
+  // cta abstraction
+  ctaText?: string;
+  // if it is a direct link
+  ctaUrl?: string;
+  ctaTarget?: string;
+  // callback when we need to take action like open pylon
+  ctaCallback?: () => void;
```

Comment:
At risk of over-abstraction, it might be cleaner to have a `cta` object to encapsulate its attributes, and it could make for better typing. But could go either way.

```
cta?: {
  type: "button" | "link";
  text: string;
  url?: string; // Only applicable if type is "link"
  target?: string; // Only applicable if type is "link"
  callback?: () => void; // Only applicable if type is "button"
};
```

Reply:
Ya I think this will be better